src/cdev/default/project.py

```diff
@@ -235,23 +256,21 @@ def update_settings_info(
         if not environment_name:
             environment_name = self.get_current_environment_name()
 
-        self._load_state()
+        self._load_info()
 
         # Remove the old environment
         previous_environment_var = [
-            x for x in self._central_state.environments if x.name == environment_name
+            x for x in self._project_info.environments if x.name == environment_name
```

Comment:
You are repeating yourself quite often by looking for the environment in the list of environments, would be a good idea to create a helper function to perform this action